src/Backoffice/Airlines/Domain/Actions/ListAirlineAction.php

```diff
@@ -14,6 +14,9 @@ class ListAirlineAction
      */
     public function execute(): LengthAwarePaginator
     {
-        return Airline::with(['flights'])->orderBy('created_at', 'desc')->paginate(10);
+        return Airline::query()
+            ->with(['flights'])
```

Comment:
```suggestion
            ->withCount('flights')
```
Since you only need the number of flights the airlines have, you should load that information instead of all the flights each airline has and then count them in each iteration. Refer to [here](https://laravel.com/docs/11.x/eloquent-relationships#counting-related-models) to learn more.